snuba/web/rpc/v1/endpoint_get_traces.py

```diff
@@ -0,0 +1,418 @@
+            condition=base_conditions_and(
+                request.meta,
+                trace_item_filters_expression,
+            ),
+            order_by=[
+                OrderBy(
+                    direction=OrderByDirection.DESC,
+                    expression=column("_sort_timestamp"),
+                ),
+            ],
+            limitby=LimitBy(limit=1, columns=[column("trace_id")]),
+            limit=request.limit if request.limit > 0 else _DEFAULT_ROW_LIMIT,
+        )
+
+        treeify_or_and_conditions(query)
+
+        results = run_query(
+            dataset=PluggableDataset(name="eap", all_entities=[]),
+            request=_build_snuba_request(request, query),
+            timer=self._timer,
+        )
+        trace_ids: dict[str, int] = {}
+        for row in results.result.get("data", []):
+            trace_ids[row["trace_id"]] = row["_sort_timestamp"]
+        return trace_ids
+
+    def _get_metadata_for_traces(
+        self,
+        request: GetTracesRequest,
+        trace_ids: dict[str, int],
+    ) -> list[GetTracesResponse.Trace]:
+        trace_item_filters_expression = trace_item_filters_to_expression(
+            request.filters[0].filter if len(request.filters) > 0 else TraceItemFilter()
+        )
+
+        selected_columns: list[SelectedExpression] = []
+        for trace_attribute in request.attributes:
+            selected_columns.append(
+                SelectedExpression(
+                    name=_ATTRIBUTES[trace_attribute.key][0],
+                    expression=_attribute_to_expression(
+                        trace_attribute,
+                        trace_item_filters_expression,
+                    ),
+                )
+            )
+
+        entity = Entity(
+            key=EntityKey("eap_spans"),
+            schema=get_entity(EntityKey("eap_spans")).get_data_model(),
+            sample=None,
+        )
+        timestamps = trace_ids.values()
+        query = Query(
+            from_clause=entity,
+            selected_columns=selected_columns,
+            condition=and_cond(
+                project_id_and_org_conditions(request.meta),
+                timestamp_in_range_condition(
+                    min(timestamps) - 2 * 3600,
+                    max(timestamps) + 2 * 3600,
+                ),
+                in_cond(
+                    f.cast(
+                        column("trace_id"),
+                        "String",
+                        alias="trace_id",
+                    ),
+                    literals_array(
+                        None, [literal(trace_id) for trace_id in trace_ids.keys()]
+                    ),
+                ),
+            ),
+            groupby=[
+                _attribute_to_expression(
+                    TraceAttribute(
+                        key=TraceAttribute.Key.KEY_TRACE_ID,
+                    ),
+                ),
+            ],
+            order_by=_convert_order_by(request.order_by),
```

Comment:
Applying the order by at this point isn't very helpful as it's only sorting the trace ids results from the first request. But i guess it'd look sorted?

I'm inclined to lean towards just
1. restrict sorts to just timestamp
2. then ensure the final results are sorted by timestamp by sorting it again in clickhouse or python

and not worry about other attributes for now